aquacomputer_d5next.c

```diff
@@ -285,32 +285,35 @@ static int aqc_get_ctrl_val(struct aqc_data *priv, int offset)
 	if (ret < 0)
 		goto unlock_and_return;
 
-	ret = get_unaligned_be16(priv->buffer + offset);
+	switch (size) {
+	case 16:
+		ret = get_unaligned_be16(priv->buffer + offset);
+		break;
+	case 8:
+		ret = priv->buffer[offset];
+		break;
+	default:
+		break;
+	}
 
 unlock_and_return:
 	mutex_unlock(&priv->mutex);
 	return ret;
 }
 
+/* Refreshes the control buffer and returns u16 value at offset */
+static int aqc_get_ctrl_val(struct aqc_data *priv, int offset)
```

Comment:
I distinctly remember groeck roasting this approach in the past when we tried (and failed) to upstream my mega-pr from hell. He wanted us to just call `return aqc_get_val(priv, offset, 16);` or `return aqc_get_val(priv, offset, 8);` directly I think